src/EditorFeatures/Core/Diagnostics/DiagnosticsClassificationTaggerProvider.cs

```diff
@@ -61,6 +61,12 @@ internal partial class DiagnosticsClassificationTaggerProvider : AbstractDiagnos
         protected internal override bool IsEnabled
             => !_editorOptionsService.Factory.GlobalOptions.GetOptionValue(DefaultTextViewHostOptions.IsInContrastModeId);
 
+        protected internal override bool SupportsDignosticMode(DiagnosticMode mode)
+        {
+            // We only support push diagnostics.  When pull diagnostics are on, diagnostic fading is handled by the lsp client.
+            return mode == DiagnosticMode.Push;
```

Comment:
@dibarbet does LSP support fading?

Reply:
yes, via the unnecessary tag - https://sourceroslyn.io/#Microsoft.CodeAnalysis.LanguageServer.Protocol/Handler/Diagnostics/AbstractPullDiagnosticHandler.cs,336